### ISSUE TITLE
Stream throws instead of calling error listener when using .map

### ISSUE THREAD

**Issue by lachenmayer** [staltz/xstream#178]
Hi there, I'm not sure if this a bug or intended behaviour.

I have a stream of things, and I'm calling a validation function (eg. `isValid`) which may throw.

This is the expected behavior:

```
thing$
  .filter(isValid)
  .addListener({next: console.log, error: console.error})
// error is printed using error listener
```

However, when I try to add a `map` operator, the error does not get passed to the error listener, as expected. The exception is not handled at all.

```diff
thing$
  .filter(isValid)
+  .map(x => {type: 'success', payload: x})
  .addListener({next: console.log, error: console.error})
// throws!
```

I've tried using other operators like `take` or another `filter`, after the initial (throwing) `filter`, and those work as expected, ie. the error listener is called.

Here is an executable example, if this is unclear: https://runkit.com/lachenmayer/xstream-error-handling

version: 10.3.0

**Comment by staltz** [comment]
Oh it seems like we missed a try catch https://github.com/staltz/xstream/blob/8f2503a4cc9ce97eadb2ba69f12a3df98b746c23/src/index.ts#L1028

Certainly a bug and certainly fixable

**Comment by staltz** [comment]
This is now fixed in xstream v10.4
